### PR TITLE
Support new BOTD#2 protocol to recognise supported content type

### PR DESCRIPTION
**Pull request by Janaka-Steph**

### CODE REVIEW

**Review by tiero — COMMENTED** [review]
prefer calling just `http` the folder

**Comment by tiero** [comment]
Also maybe makes sense having them under `api-spec/openapi/swagger`? Like we do for server-side stubs

**Comment by altafan** [comment]
let's add an httpClient wrapper that sticks with `TraderClientInterface` and calls the methods of the client generated by swagger under the hood.

**Comment by Janaka-Steph** [comment]
Seems more tricky actually. The types are not exactly the same.
```
proposeTrade(
  market: types.Market,
  type: types.TradeType,
  swapRequest: Uint8Array
): Promise<Uint8Array> {
  const res = await this.client.tradeServiceProposeTrade({
    market,
    type: type as unknown as V1TradeType,
    swapRequest: swapRequest as V1SwapRequest
  })
}
```
Need to do ugly cast like this everywhere
I guess it's better to just not auto generate http stubs

**Comment by altafan** [comment]
@tiero any suggestion?

**Comment by tiero** [comment]
Not super expert of Swagger code generation. 

I think for 4 API should be fine to maintain them manually

**Comment by tiero** [comment]
just btw found this
https://github.com/drwpow/openapi-typescript

**Comment by Janaka-Steph** [comment]
> just btw found this https://github.com/drwpow/openapi-typescript

I tried it but not convinced by the result.
`npx openapi-typescript "/Users/stephane/Documents/SevenLabs/tdex/daemon/tdex-daemon/api-spec/openapi/swagger/tdex/v1/**/*.json" --output schemas/`

**Comment by Janaka-Steph** [comment]
@tiero @altafan Is something like that enough https://github.com/tdex-network/tdex-sdk/blob/cc21392906ca5097931f86a9db333a049d397736/src/api-spec/openapi/swagger/trade2.ts ?

**Comment by tiero** [comment]
> @tiero @altafan Is something like that enough https://github.com/tdex-network/tdex-sdk/blob/cc21392906ca5097931f86a9db333a049d397736/src/api-spec/openapi/swagger/trade2.ts ?

did you tested? Because the blinding keys we pass must be serialized in a specific way and dont thinl JSON.stringify will do the right thing. AFAIK base64 right @altafan ?

**Comment by Janaka-Steph** [comment]
> > @tiero @altafan Is something like that enough https://github.com/tdex-network/tdex-sdk/blob/cc21392906ca5097931f86a9db333a049d397736/src/api-spec/openapi/swagger/trade2.ts ?
> 
> did you tested? Because the blinding keys we pass in the swap message must be serialized in a specific way and dont thinl JSON.stringify will do the right thing. AFAIK base64 right @altafan ?

No I didn't test it

**Comment by tiero** [comment]
Yeah, also SwapRequest.fromBinary is not doing the right thing I guess, you need to have the right payload destrcutured

File: src/api-spec/openapi/swagger/trade2.ts [hunk 1]
```diff
@@ -12,28 +13,24 @@ export class TradeServiceHttp implements TraderClientInterface {
...
+    console.log('this.providerUrl', this.providerUrl);
+    const response = await axios.post(this.providerUrl + path, {
         market: market,
         type: tradeType.toString(),
-        swapRequest: SwapRequest.fromBinary(swapRequestSerialized),
-      }),
-    })
-    const str = await response.json()
+        swapRequest: SwapRequest.toJsonString(SwapRequest.fromBinary(swapRequestSerialized)),
+        //swapRequest: Buffer.from(swapRequestSerialized).toString('base64'),
```
Comment (tiero) [inline src/api-spec/openapi/swagger/trade2.ts:31]: don think this work either:

we should pass each field of the swap request, one by one, for the type that swagger defintion specify.

Only thing to keep into consideration that `bytes` fields are passed as `string` in JSON, which means base64

Comment (tiero) [reply src/api-spec/openapi/swagger/trade2.ts]:
```json
{
  "market": {
    "base_asset": "Hello",
    "quote_asset": "Hello"
  },
  "type": 0,
  "swap_request": {
    "id": "ec033502-fbe2-4ab6-a455-f5711aa749ba",
    "amount_p": 100,
    "asset_p": "Hello",
    "amount_r": 100,
    "asset_r": "Hello",
    "transaction": "Hello",
    "input_blinding_key": {
      "script": "base64BlindingKey" 
    },
    "output_blinding_key": {
      "script": "base64BlindingKey" 
    }
  }
}
```

Comment (altafan) [reply src/api-spec/openapi/swagger/trade2.ts]: In general, though, this test must be dropped.

We should just write a new one in `test/trade.integration.test.ts` where we don't use the default client, but the new one defined in `openapi/swagger`.

**Pull request closed without merging**